mllib/src/main/scala/org/apache/spark/ml/tree/treeModels.scala

```diff
@@ -58,3 +58,43 @@ trait DecisionTreeModel {
     header + rootNode.subtreeToString(2)
   }
 }
+
+/**
+ * :: AlphaComponent ::
+ *
+ * Abstraction for models which are ensembles of decision trees
+ *
+ * TODO: Add support for predicting probabilities and raw predictions
+ */
+@AlphaComponent
+trait TreeEnsembleModel {
+
+  // Note: We use getTrees since subclasses of TreeEnsembleModel will store subclasses of
+  //       DecisionTreeModel.
+
+  /** Trees in this ensemble. Warning: These have null parent Estimators. */
+  def getTrees: Array[DecisionTreeModel]
```

Comment:
If we don't allow users to set trees, it is a property and maybe calling it `trees` would be sufficient. Same for `getTreeWeights`.